### PR TITLE
Fix Geonovum #182: determine level in TOC for large numbers of subsec…

### PR DESCRIPTION
**Pull request by thijsbrentjens**
Fix calc of level for a section in the TOC. See issue #182

### CODE REVIEW

**Comment by marcoscaceres** [comment]
I wonder if this also affects ReSpec upstream too (given that it's a core module)... @sidvishnoi ^^^

**Comment by thijsbrentjens** [comment]
That was the next thing on my list :) (and sorry I was not able to do things for a long time, but I have been occupied by other things)

**Comment by sidvishnoi** [comment]
Yes, it does affect upstream. Good find!

**Comment by thijsbrentjens** [comment]
@sidvishnoi shoudld I create a PR for that too? I'll make it via my personal fork then, that's a bit cleaner

File: src/core/structure.js [hunk 1]
```diff
@@ -68,7 +68,9 @@ function scanSections(sections, maxTocLevel, { prefix = "" } = {}) {
       : appendixMode
       ? alphabet.charAt(index - lastNonAppendix)
       : prefix + index;
-    const level = Math.ceil(secno.length / 2);
+    // Thijs Brentjens, Geonovum: fix for determining the level:
+    // count the number of dots in the section number, to determine the level
+    const level = (secno.match(/\./g)||[]).length + 1;
```
Comment (sidvishnoi) [inline src/core/structure.js]: Maybe use this?
```suggestion
    const level = secno.split(".").length;
```
(It's unexpectedly [faster](https://perf.link/#eyJpZCI6ImFqZnFzMHIwbzg2IiwidGl0bGUiOiJGaW5kaW5nIG51bWJlcnMgaW4gYW4gYXJyYXkgb2YgMTAwMCIsImJlZm9yZSI6InN0ciA9IFwiMS4yLjIxLjEyXCIiLCJ0ZXN0cyI6W3sibmFtZSI6IiIsImNvZGUiOiJzdHIuc3BsaXQoXCIuXCIpLmxlbmd0aCIsInJ1bnMiOlsyNTAwMDAsMzgzMDAwLDI5MDUwMCw0OTk1MDAsNDUxNTAwLDk5MDAwLDEyMTAwMCwzMTE1MDAsODQ4MDAwLDcwNzUwMCwxNzA1MDAsMjg3MDAwLDU4ODUwMCwzMjc1MDAsODM1MDAsMzU2MDAwLDkwNTAwLDQzNzAwMCw0OTk1MDAsNTI1NTAwLDU3MDUwMCw0MzIwMDAsNTE0MDAwLDM2ODAwMCw1MzcwMDAsNDk5NTAwLDI3NzUwMCwxNTYwMDAsNDk5NTAwLDE4NjUwMCw1MDAwMCwxNzk1MDAsMzA1NTAwLDU2MzUwMCw0OTk1MDAsMzcyMDAwLDU1MDAsMjA1MDAwLDI0MDUwMCwyNDMwMDAsNTEyNTAwLDI1MDAwLDIwMTAwMCw1MTUwMCwyNjc1MDAsNTAwMDAsNTYyNTAwLDIxNDUwMCw1NTAwLDI3MDUwMCwzMDAwLDMwMTAwMCw0OTUwMCw2MzAwMCwxNTUwMDAsMzk4NTAwLDE1OTAwMCw1NTAwLDQ5MzUwMCwyNTU1MDAsNTA5MDAwLDMwNTUwMCw2MTYwMDAsMjgwMDAsMzUwNTAwLDUzMDAwMCwzMzEwMDAsMjg5NTAwLDUxMzAwMCw1MTIwMDAsNDk5NTAwLDQ5OTUwMCwzNjAwMDAsNTA2NTAwLDQ5OTUwMCwzMDYwMDAsNjQxNTAwLDU1MDAsNTAwMDAsMjg4NTAwLDQ5OTUwMCwxNDc1MDAsMTk1MDAsMjk2MDAwLDE3NTAwMCw0OTk1MDAsMjIzNTAwLDQ5MDAwLDM0NjAwMCw0ODAwMCwyNzUwMDAsMzg3NTAwLDMwMDAsNjQ1MDAsMjgwMDAwLDI1MDAwMCwxNzAwMCw2NzY1MDAsNTAwMDAsMjk2MDAwXSwib3BzIjozMDMxOTV9LHsibmFtZSI6IiIsImNvZGUiOiIoc3RyLm1hdGNoKC9cXC4vZykgfHwgW10pLmxlbmd0aCArIDEiLCJydW5zIjpbNTAwMDAsMTAwNTAwLDY3NTAwLDE0ODAwMCwyMDU1MDAsMTk1MDAsMjkwMDAsNzcwMDAsMjgyMDAwLDI4MjAwMCw0MTAwMCw0MTAwMCwyMDI1MDAsNTUwMCwxODUwMCw4NTAwMCwyNDAwMCwxMDA1MDAsMTA1MDAwLDE1MjUwMCwxNzA1MDAsMjAxMDAwLDE5OTUwMCw5NzUwMCwyNTAwMCwxODM1MDAsNTMwMDAsMzE1MDAsMjA0MDAwLDQzNTAwLDEwMDAwLDEyODUwMCw2NTAwMCwxOTcwMDAsMjE4NTAwLDEwMDUwMCw0NTAwLDUwMDAwLDEwMDAwLDI1MDAwLDIwMTUwMCw0NTAwLDc1MDAsMTEwMDAsNTAwMDAsNjAwMCwxODEwMDAsNDk1MDAsMTUwMCw1MDAwMCwxNTAwLDY0NTAwLDgwMDAsNDkwMDAsMzEwMDAsMTAwNTAwLDM4MDAwLDM4NTAwLDEzNTUwMCw2MjAwMCwxNTUwMDAsNzMwMDAsMjExNTAwLDU1MDAsMTAwNTAwLDE5NzUwMCwxMDA1MDAsNTk1MDAsMTM1NTAwLDE1NzUwMCwxNDE1MDAsNDkwMDAsMTAwNTAwLDE2MTAwMCwxMDE1MDAsNjkwMDAsMjgxMDAwLDI1MDAsMTA1MDAsOTgwMDAsMTY0NTAwLDE1MDAwLDUwMDAsNTM1MDAsMzEwMDAsODYwMDAsNDkwMDAsNzAwMCwxMDA1MDAsNjUwMCw3NDUwMCwxMDA1MDAsNDcwMDAsMTYwMDAsMzUwMDAsNTIwMDAsNTAwMCwyNDA1MDAsMTAwMDAsNjE1MDBdLCJvcHMiOjg1MTYwfV0sInVwZGF0ZWQiOiIyMDIwLTA1LTE4VDE1OjAzOjM3LjI5MFoifQ%3D%3D) too :smile: )

**Comment by sidvishnoi** [comment]
> should I create a PR for that too?

That would be great!

File: src/core/structure.js (hunk 1 above)
Comment (thijsbrentjens) [reply src/core/structure.js]: Hmm, I clicked on something accidentally. Your change is simpler, better. I'll use that

**Review by lvdbrink — APPROVED** [review]
Tested and working with one of the docs that demonstrated the problem.

**Pull request merged**